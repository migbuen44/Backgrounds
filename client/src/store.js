import { configureStore } from '@reduxjs/toolkit';
import userLoginReducer from './slices/userLoginSlice';
import loginModalReducer from './slices/loginModalSlice';
import userInfoReducer from './slices/userInfoSlice';

export default configureStore({
  reducer: {
    userLogin: userLoginReducer,
    loginModal: loginModalReducer,
    userInfo: userInfoReducer,
  },
});
