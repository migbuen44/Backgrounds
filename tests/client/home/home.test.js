import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer'
import store from '../../../client/src/store';
import Home from '../../../client/src/components/home/home';

describe('Home', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <Home />
      </Provider>, div);
  })

  it('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
});