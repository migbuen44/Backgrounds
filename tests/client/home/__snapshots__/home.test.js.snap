// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
Array [
  <div
    className="lib-images-module-toggle-2tKw"
    onClick={[Function]}
  />,
  <div
    className="lib-images-module-imageContainer-hYn6 scroll"
  />,
  <a
    className="lib-spotifyLogin-module-spotifyLoginButton-2601"
    href="https://accounts.spotify.com/authorize?client_id=4cb458aef9d344d2a58c62e7da3d0da5&response_type=code&redirect_uri=https://backgroundsmb.com/&scope=streaming%20user-read-email%20user-read-private%20user-library-read%20user-library-modify%20user-read-playback-state%20user-modify-playback-state"
  >
    Spotify Login
  </a>,
  <div
    className="lib-search-module-search-3DLX"
  >
    <form
      className="lib-search-module-searchContainer-o4DC"
      onSubmit={[Function]}
    >
      <input
        className="lib-search-module-searchBar-1TRu"
        onChange={[Function]}
        placeholder="Search..."
      />
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-search fa-w-16 lib-search-module-searchIcon-1VCs"
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </form>
  </div>,
  <div
    className="lib-icons_and_holders-module-loginContainer-1qYL iconCircle click"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-user fa-w-14 lib-icons_and_holders-module-loginButton-1mCw"
      data-icon="user"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={
        Object {
          "color": "black",
        }
      }
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>,
  <div
    className="lib-home-module-saveSelectContainer-CIYC iconCircle click"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-bookmark fa-w-12 "
      data-icon="bookmark"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={
        Object {
          "color": "black",
        }
      }
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>,
  <div
    className="lib-home-module-musicPlayerHolder-1Gxf"
  >
    <div
      className="lib-home-module-playerHolderMessage-2MYH"
    >
      Sign in to Spotify to play songs
    </div>
  </div>,
]
`;
